allennlp/training/metrics/bleu.py

```diff
@@ -116,10 +118,21 @@ def __call__(
         None
         """
         predictions, gold_targets = self.detach_tensors(predictions, gold_targets)
+        device = gold_targets.device
+        if is_distributed():
+            world_size = dist.get_world_size()
+
         for ngram_size, _ in enumerate(self._ngram_weights, start=1):
             precision_matches, precision_totals = self._get_modified_precision_counts(
                 predictions, gold_targets, ngram_size
             )
+            if is_distributed():
+                _precision_matches = torch.tensor(precision_matches).to(device)
+                _precision_totals = torch.tensor(precision_totals).to(device)
+                dist.all_reduce(_precision_matches, op=dist.ReduceOp.SUM)
+                dist.all_reduce(_precision_totals, op=dist.ReduceOp.SUM)
+                precision_matches = _precision_matches.item() / world_size
+                precision_totals = _precision_totals.item() / world_size
```

Comment:
Aren't `precision_matches` and `precision_totals` counts? Why is it OK to average them?

Reply:
That's correct. Fixing.